cos_alerter/daemon.py

```diff
@@ -40,19 +18,17 @@ def sigint(_, __):
 
 def main():
 
-    # Create the initial data file
-    with open(config['watch']['data_file'], 'w') as f:
-        json.dump({'alert_time': time.time(), 'notify_time': 0.0}, f)
+    AlerterState.initialize()
 
     signal.signal(signal.SIGINT, sigint)
     signal.signal(signal.SIGUSR1, send_test_mail)
 
     subprocess.Popen(['waitress-serve', 'cos_alerter.server:app'])
 
     while(True):
-        with AlerterState() as time_data:
-            if time.time() - time_data.alert_time > down_interval():
-                notify(time_data)
+        with AlerterState() as state:
```

Comment:
Is instantiating and disposing the alerter state on every loop intentional? Seems a little odd to me.

Reply:
Yes because when you create the `AlerterState `, it acquires a lock on your state file. If we hold that lock forever, the web server can never write to it.

Reply:
Well, when you use it as a contextmanager. In practice, we could probably explicitly lock/unlock with a dedicated attribute/method or use `@contextlib.contextmanager` on some method, but it'd probably be really clear by just renaming `AlerterState` to `StateFutex` or something.

Reply:
I changed the code to only instantiate once. Does this look better?